nexus/types/src/external_api/params.rs

```diff
@@ -1000,9 +1000,9 @@ pub struct InstanceCreate {
     #[serde(default)]
     pub disks: Vec<InstanceDiskAttachment>,
 
-    /// List of public keys (identified by ID or name) to be associated with the instance.
+    /// List of public SSH keys (identified by ID or name) to be associated with the instance.
     #[serde(default)]
-    pub public_keys: Option<Vec<NameOrId>>,
+    pub ssh_keys: Option<Vec<NameOrId>>,
```

Comment:
I think I like the field to specify that the keys are public (purely from a security point of view), how about `public_ssh_keys` instead?

Reply:
If you're going to do that, it should be **ssh_public_keys** probably.  **Public Key** is the thing and SSH is a qualifier.

Reply:
I like that, I like having both adjectives in there.

Reply:
Something I talked through with with @charliepark too. I think whatever we do, it's worth keeping the resource label consistent across usages.

Reply:
I'd note that GitHub just calls them SSH Keys. I think (hope) the _public_ aspect is implied and that folks aren't in the habit of submitting private keys. 

I think we could go further than this and augment our clients to _explicitly_ reject keys formatted like a private key. They typically have beginning/ending tags of various forms and we can reject based on the presence of that.

Reply:
> I'd note that GitHub just calls them SSH Keys. I think (hope) the public aspect is implied and that folks aren't in the habit of submitting private keys.

Is there any harm in being explicit and calling them `ssh_public_keys`? As far as I'm aware we're expecting to have users with varying degree of technical expertise. Being explicit can be helpful.

Reply:
The rename touches a lot of files so I've pulled it out into a separate issue to be done in a follow up PR: https://github.com/oxidecomputer/omicron/issues/4866